include/oc_push.h

```diff
@@ -0,0 +1,111 @@
+/****************************************************************************
+ *
+ * Copyright 2021 ETRI All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * Created on: Aug 23, 2021,
+ * 				Author: Joo-Chul Kevin Lee (rune@etri.re.kr)
+ *
+ *
+ ****************************************************************************/
+
+#ifndef OC_PUSH_H
+#define OC_PUSH_H
+
+#include <stdio.h>
+
+#include "oc_config.h"
+#include "oc_helpers.h"
+#include "oc_rep.h"
+#include "oc_ri.h"
+#include "oc_endpoint.h"
+#include "port/oc_log.h"
+#include "util/oc_memb.h"
+#include "util/oc_process.h"
+
+
+#if defined(OC_PUSHDEBUG) || defined(OC_DEBUG)
+#define OC_PUSH_DBG(...) OC_LOG("D", __VA_ARGS__)
+#define OC_PUSH_WRN(...) OC_LOG("W", __VA_ARGS__)
+#define OC_PUSH_ERR(...) OC_LOG("E", __VA_ARGS__)
+#else
+#define OC_PUSH_DBG(...)
+#define OC_PUSH_WRN(...)
+#define OC_PUSH_ERR(...)
+#endif
+
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+
+#define PUSHCONF_RSC_PATH "/pushconfig"
+#define PUSHCONF_RSC_TYPE "oic.r.pushconfiguration"
+#define PUSHCONF_RSC_NAME "Push Configuration"
+
+#define PUSHRECVS_RSC_PATH "/pushreceivers"
+#define PUSHRECVS_RSC_TYPE "oic.r.pushreceiver"
+#define PUSHRECVS_RSC_NAME "Push Receiver Configuration"
+
+
+
+/**
+ * @brief object used to store Resource pushed to "oic.r.pshreceiver:receivers[i].receiveruri"
+ */
+typedef struct oc_pushd_rsc_rep
+{
+	struct oc_pushd_rsc_rep *next;
+	oc_resource_t *resource;        ///< used to point any pushed Resource managed by iotivity-lite
+	oc_rep_t *rep;                  ///< payload of pushed Resource
+} oc_pushd_rsc_rep_t;
+
+
+/**
+ * @brief string representing response code
+ */
+extern const char *cli_status_strs[];
```

Comment:
We should avoid global variables if possible:

1) can't cli_status_strs and cli_statusstr be moved to oc_push.c? I don't see them being used outside of that file. If they need to be in a header then they shouldn't be in a public header (eg. a header in the include directory, these headers are copied to /usr/local/lib/iotivity-lite when somebody install the library and thus should feature only public classes and functions).

2) oc_push_arrived -> make the declaration of the variable static in the oc_push.c and create a setter function:
```C
typedef void (*oc_on_push_arrived_t)(oc_pushd_rsc_rep_t *);

void oc_set_on_push_arrived(oc_on_push_arrived_t*);
```

(or something like that), which sets the variable.

Reply:
1. I'm using that macro in other apps (e.g. push configurator)
2. fixed

Reply:
1. Would a public function work for you? Enum `oc_status_t` is defined in `oc_ri.h`, when I look into that header there is already a public function called `oc_status_code` which converts the status to some other code. Lets add a new public function `oc_status_to_str(oc_status_t)` after `oc_status_code` and you can use that in your app.

Reply:
done